features/ingredients_api/router.py

```diff
@@ -0,0 +1,134 @@
+def get_ingredient(ingredient_id: int = fastapi.Path()):
+
+    """Get ingredient
+
+    :param ingredient_id: Identifier for ingredient
+    :type int
+    ...
+    :raises HTTPException: Raised if ingredient with ingredient_id not found
+    ...
+    :return: Found ingredient
+    :rtype: Ingredient
+    """
+
+    try:
+        return features.ingredients_api.operations.get_ingredient(ingredient_id)
+    except features.ingredients_api.exceptions.IngredientNotFoundException:
+        raise fastapi.HTTPException(
+            status_code=fastapi.status.HTTP_404_NOT_FOUND,
+            detail=f"Ingredient with {ingredient_id=} not found"
+        )
+
+@ingredients_router.post('/', response_model=Ingredient)
+def create_ingredient(create_ingredient_input_model: CreateIngredientInputModel):
+
+    """Create ingredient
+
+        :param : name of ingredient
+        :type str
+        ...
+        :param : category of ingredient
+        :type str
+        ...
+        :param : calories of ingredient
+        :type float
+        ...
+        :param : carbo of ingredient
+        :type float
+        ...
+        :param : fats of ingredient
+        :type float
+        ...
+        :param : protein of ingredient
+        :type float
+        ...
+        :param : cholesterol of ingredient
+        :type float
+        ...
+        :param : measurement of ingredient
+        :type float
+        ...
+        ...
+        :raises HTTPException: Raised if ingredient with the same name already exists
+        ...
+        :return: Created ingredient
+        :rtype: Ingredient
+        """
+
+    try:
+        return features.ingredients_api.operations.create_ingredient(create_ingredient_input_model.name)
+
+    except features.ingredients_api.exceptions.IngredientNameViolationException:
+
+        raise fastapi.HTTPException(
+            status_code=fastapi.status.HTTP_404_NOT_FOUND,
+            detail=f"Ingredient with name {create_ingredient_input_model.name} already exists"
+        )
+
+@ingredients_router.patch('/{ingredient_id', response_model=Ingredient)
+def update_ingredient(ingredient_id: int = fastapi.Path(), patch_ingredient_input_model: PatchIngredientInputModel = fastapi.Body()):
+
+    """Update ingredient
+
+    :param ingredient_id: Identifier for ingredient
+    :type int
+    ...
+    :raises HTTPException: Raised if ingredient with ingredient_id not found
+    ...
+    :return: Updated ingredient
+    :rtype: Ingredient
+    """
+
+    try:
+        return features.ingredients_api.operations.update_ingredient(ingredient_id=ingredient_id, **patch_ingredient_input_model.model_dump())
+
+    except features.ingredients_api.exceptions.IngredientNameViolationException:
+
+        raise fastapi.HTTPException(
+            status_code=fastapi.status.HTTP_400_BAD_REQUEST
+        )
+
+@ingredients_router.patch('/{ingredient_id', response_model=Ingredient)
+def soft_delete_ingredient(ingredient_id: int = fastapi.Path()):
```

Comment:
it should be delete request
and name of the fucntion should be just delete_ingredient